### PR TITLE
DEVX-2461: prevent automatic approval of third party templates

### PR DESCRIPTION
**Pull request by AmyJZhao**
## Description

[DEVX-2461](https://jira.corp.adobe.com/browse/DEVX-2461)

Adobe templates (NPM package stored in @adobe org) can be automatically approved after passing checks. Third party templates should have a status "InVerification" to get manual approval from Adobe team.

## How Has This Been Tested?

Adobe template: 

- add-template: https://github.com/devx-services/aio-template-submission/issues/38
- update-template: https://github.com/devx-services/aio-template-submission/issues/41

Third party template: 

- https://github.com/devx-services/aio-template-submission/issues/39
- Approved template updated: https://github.com/devx-services/aio-template-submission/issues/45

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by roribio — APPROVED** [review]

File: src/update-template.js [hunk 1]
```diff
@@ -55,6 +54,13 @@ const { TEMPLATE_STATUS_APPROVED } = require('../src/registry');
         }
 
         const savedTemplate = getFromRegistry(npmPackageMetadata.name);
+        if(adobeRecommended || savedTemplate["status"] == TEMPLATE_STATUS_APPROVED) {
```
Comment (slitviachenko) [inline src/update-template.js:57]: @AmyJZhao , we shouldn't use `adobeRecommended` for deciding if a template should be approved or not. Logic for setting up `adobeRecommended` might be changed, and that shouldn't affect the approval process. We could use a previous logic with checking a number of _stars_, a 3rd-party repo might have many stars, but that doesn't mean that we should approve that template automatically.

File: src/add-to-registry.js [hunk 2]
```diff
@@ -36,13 +36,16 @@ const { isInRegistry, addToRegistry, getFromRegistry, updateInRegistry, TEMPLATE
...
             "adobeRecommended": adobeRecommended,
             "keywords": npmPackageMetadata.keywords,
-            "status": TEMPLATE_STATUS_APPROVED,
             "links": {
                 "npm": npmUrl,
                 "github": gitHubUrl
             }
         }
-
+        if(adobeRecommended) {
```
Comment (slitviachenko) [inline src/add-to-registry.js:44]: @AmyJZhao , we shouldn't use `adobeRecommended` for deciding if a template should be approved or not. Logic for setting up `adobeRecommended` might be changed, and that shouldn't affect the approval process. We could use a previous logic with checking a number of _stars_, a 3rd-party repo might have many stars, but that doesn't mean that we should approve that template automatically.

File: src/update-template.js [hunk 3]
```diff
@@ -34,13 +34,12 @@ const { TEMPLATE_STATUS_APPROVED } = require('../src/registry');
...
             "adobeRecommended": adobeRecommended,
             "keywords": npmPackageMetadata.keywords,
-            "status": TEMPLATE_STATUS_APPROVED,
             "links": {
                 "npm": npmUrl,
                 "github": gitHubUrl
             }
         }
-
+
```
Comment (slitviachenko) [inline src/update-template.js:42]:
```suggestion

```
TODO: need to add a linter with appropriate rules

**Review by slitviachenko — CHANGES_REQUESTED** [review]
@AmyJZhao , we shouldn't use `adobeRecommended` for deciding if a template should be approved or not. Logic for setting up `adobeRecommended` might be changed, and that shouldn't affect the approval process. We could use a previous logic with checking a number of _stars_, a 3rd-party repo might have many stars, but that doesn't mean that we should approve that template automatically.
@AmyJZhao , you created a new `needs-more-verification-third-party` job, and that is right, it should encapsulate all logic related to deciding if an additional verification step is needed or not, we can introduce some allow list for trusted Adobe partners, for example, so we should have only one place to change. Maybe we need to create a reusable Github workflow, as we did for [checking admin permissions](https://github.com/adobe/aio-template-submission/blob/main/.github/workflows/check-permissions-workflow.yml).

@roribio , ^^^ your thoughts ?

File: .github/workflows/update-template.yml [hunk 4]
```diff
@@ -87,10 +88,22 @@ jobs:
...
+        with:
+          message: |
+            The template has been updated in Template Registry. However, templates by 3rd party developers need further validation from the Adobe team before full approval. Feedback will be provided soon.
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   success-close-issue:
     name: Close issue
     runs-on: ubuntu-latest
-    needs: [update-template]
+    needs: [update-template, parse-issue-form-body]
+    if: ${{ startsWith(needs.parse-issue-form-body.outputs.npm-package, '@adobe/') || needs.update-template.outputs.is-approved == 'true'}}
```
Comment (slitviachenko) [inline .github/workflows/update-template.yml:106]: Please use `needs-more-verification-third-party`'s _decision_ here.

File: .github/workflows/add-template.yml [hunk 5]
```diff
@@ -60,10 +60,22 @@ jobs:
...
+        with:
+          message: |
+            @${{ github.event.issue.user.login }}, thank you for your submission. The Adobe team will validate your template and provide our feedback shortly.
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   success-close-issue:
     name: Close issue
     runs-on: ubuntu-latest
-    needs: [add-to-registry]
+    needs: [add-to-registry, parse-issue-form-body]
+    if: ${{ startsWith(needs.parse-issue-form-body.outputs.npm-package, '@adobe/') }}
```
Comment (slitviachenko) [inline .github/workflows/add-template.yml:78]: Please use `needs-more-verification-third-party`'s _decision_ here.

File: src/update-template.js (hunk 1 above)
Comment (roribio) [reply src/update-template.js:57]: I agree. From the conversations I've heard, there's still some debate on what is an "Adobe Recommended" template.

**Pull request closed without merging**